src/wrappers/WebWrapperUtils.ts

```diff
@@ -0,0 +1,218 @@
+import InvocationPlugin from '../plugins/Invocation';
+import TracePlugin from '../plugins/Trace';
+import LogPlugin from '../plugins/Log';
+import ExecutionContextManager from '../context/ExecutionContextManager';
+import Reporter from '../Reporter';
+import ThundraLogger from '../ThundraLogger';
+import PluginContext from '../plugins/PluginContext';
+import ConfigProvider from '../config/ConfigProvider';
+import ThundraConfig from '../plugins/config/ThundraConfig';
+import ConfigNames from '../config/ConfigNames';
+import { ApplicationManager } from '../application/ApplicationManager';
+import ExecutionContext from '../context/ExecutionContext';
+import Utils from '../utils/Utils';
+import ThundraTracer from '../opentracing/Tracer';
+import * as asyncContextProvider from '../context/asyncContextProvider';
+import * as opentracing from 'opentracing';
+import MonitoringDataType from '../plugins/data/base/MonitoringDataType';
+import InvocationData from '../plugins/data/invocation/InvocationData';
+import TimeoutError from '../error/TimeoutError';
+import InvocationSupport from '../plugins/support/InvocationSupport';
+import InvocationTraceSupport from '../plugins/support/InvocationTraceSupport';
+import { HttpTags, DomainNames, ClassNames, SpanTags, TriggerHeaderTags } from '../Constants';
+import ThundraSpanContext from '../opentracing/SpanContext';
+
+const get = require('lodash.get');
+
+export default class WrapperUtils {
+    static beforeRequest(request: any, plugins: any[]) {
```

Comment:
I think it is better to make `beforeRequest` async and await `beforeInvocation` calls of plugins as `beforeInvocation` phase might be async for some plugins such as metric. Even though currently metric plugin is not available for web wrapper, I think we should contact with plugins in async way.